crates/ide/src/lib.rs

```diff
@@ -148,7 +148,7 @@ pub struct AnalysisHost {
 }
 
 impl AnalysisHost {
-    pub fn new(lru_capacity: Option<usize>) -> Self {
+    pub fn new(lru_capacity: Option<usize>) -> AnalysisHost {
```

Comment:
Just curious: is there a reasoning to prefer type name rather than `Self`?

Reply:
I find this version more readable -- no need to do a type dereference in my head